pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -81,10 +82,23 @@ func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 		// because any new CSINode could make pods that were rejected by CSI volumes schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
 		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
-		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}},
+		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add}, QueueingHintFn: pl.isSchedulableAfterPVCAdded},
 	}
 }
 
+func (pl *CSILimits) isSchedulableAfterPVCAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, addedPvc, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPVCAdded: %w", err)
+	}
+
+	if addedPvc.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	return framework.Queue, nil
```

Comment:
If the PVC is not used by this Pod, the event may not make the Pod become scheduable.

cc @sanposhiho

Reply:
@utam0k I believe it's in the acceptable range of complexity if we add a check ^.

Reply:
Make sense +1

a8a60bbc5e7
d292b7c785c